src/MapView/MapView.tsx

```diff
@@ -1,37 +1,35 @@
-import {forwardRef, useEffect, useImperativeHandle, useMemo, useRef} from 'react';
-import Mapbox, {MarkerView} from '@rnmapbox/maps';
+import {forwardRef, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import Mapbox, {MapState, MarkerView} from '@rnmapbox/maps';
 import {View} from 'react-native';
 import {MapViewProps, MapViewHandle} from './MapViewTypes';
 import Direction from './Direction';
 import Utils from './utils';
 
 const MapView = forwardRef<MapViewHandle, MapViewProps>(function MapView(
-    {accessToken, style, styleURL, pitchEnabled, mapPadding, initialState, waypoints, directionCoordinates, directionStyle},
+    {accessToken, style, styleURL, pitchEnabled, mapPadding, initialState, waypoints, directionCoordinates, directionStyle, isFocused},
```

Comment:
I actually don't like the fact that I have to pass the screen state of the parent component to `MapView`

Is there any way we can see if this MapView component is focused/visible or not from inside the component without passing the prop?

Reply:
@hayata-suenaga Initially I used `useIsFocused`  hook from   `@react-navigation/native` to get IsFocused state, inside MapView component.  In that case we needed to add new dependency to `react-native-x-maps` library.  

Since MapView is library, it can be used out of navigation context,  which is required for useIsFocused hook, or can be used inside app where using another react-native navigation library.

That's why I decided to pass isFocused state from parent.

Reply:
Good question. When we adding or changing the address on the `Waypoint Editor` screen, the waypoints are changing before Map screen becomes focused, and we could not animate map camera to show map markers and directions. Idle state needed for starting new transition (camera position change) because I found that when we have intersection of transitions, camera is not working properly.

Reply:
okay I understand we're actually planning to move the code from this library to NewDot in the future. So a little bit of mix of concerns is acceptable at this point 👍